lit/docs/getting-started/resources.lit

```diff
@@ -0,0 +1,542 @@
+  Linux tarball you can \link{download from
+  Github}{https://github.com/concourse/concourse/releases}. You'll notice
+  that the Linux tarball is much larger than the macOS or Windows tarball
+  because of all the bundled resources.
+
+  \aside{
+      You can find out which resources a worker has by running:
+      \codeblock{}{{
+      $ fly -t tutorial workers --details
+      }}
+  }
+
+  Resources only run on Linux workers because resources are distributed as
+  Linux container images. There are currently no resources for macOS or
+  Windows. Only \reference{task-step}{task steps} can run on macOS or Windows
+  workers.
+}
+
+\section{
+  \title{Versions}
+
+  Resources represent the external system or object to Concourse by emitting
+  \reference{schema.version}{\code{versions}}. When a new version is emitted
+  by a resource, that is how Concourse knows to start trigger jobs.
+
+  A version is any way that a resource can uniquely identify the state of the
+  external system or object.
+
+  For example, the git resource emits versions based on the SHA of new commits
+  it finds. A single version from the git resource will look like this to
+  Concourse.
+
+  \codeblock{json}{{{
+  { "ref": "04194bfc880c457a5b00f07db78d0532620414cc" }
+  }}}
+
+  Let's start digging into resources a bit more by going over the resource interface.
+}
+
+\section{
+  \title{Resource Interface}
+
+  Resources are \link{container
+  images}{https://en.wikipedia.org/wiki/List_of_Linux_containers} that contain
+  three executables. Each executable is run by a different type of
+  \reference{steps}{step} within a pipeline:
+
+  \list{
+      \code{/opt/resource/check} - implicitly ran when a
+      \reference{jobs}{job} contains a \reference{get-step}{get step}.
+      Should return the latest version from the external system or object.
+      Its responsibility is to find new versions.
+  }{
+      \code{/opt/resource/in} - run in a \reference{get-step}{get step}.
+      \code{in} is given a specific version (generated by a \code{check} or
+      \reference{put-step}{\code{put} step}) and retrieves the files
+      representing that version from the external system or object.
+  }{
+      \code{/opt/resource/out} - run in a \reference{put-step}{put step}.
+      Generates a new version, usually based on some input generated by
+      another step in the job. Depending on the resource, this may mean
+      sending something to the external system. For the git resource, this
+      means pushing commits to the external git repository.
+  }
+
+  That's a high-level overview of the resource interface. Next, we will learn
+  how to add resources to a pipeline.
+}
+
+\section{
+  \title{Get Steps}
+
+  Let's add some automation to our \code{hello-world.yml} pipeline and have
+  it \bold{trigger every one minute}. We will use the \link{time
+  resource}{https://github.com/concourse/time-resource/} to accomplish this.
```

Comment:
IMO the most easiest resource type to explain resources is the `git` resource, since most developers are familiar with git and the commit model to some extent. The `time` resource, on the other hand, feels a little less natural - sure, time *is* external state, but:
1. I suspect most people don't think of it in the same way as e.g. a `git` repo, and
2. We're currently working to get rid of it as a resource type (in favour of a var source), so I feel like introducing it as a resource only to change the mental model shortly thereafter is a bit confusing

So, maybe we could point to a sample git repository instead? I think that'd help people picture real-world use cases of Concourse automation. For explaining the `put` step, we could suggest they create a fork of the repo and push to that?

Reply:
git would be my goto explanation as well. I just dislike the overhead of having to tell someone to setup a git repo 😓 
Anyway, I rewrote the entire section using git instead cause it is better 😁